jazz_codeq/utils.js

```diff
@@ -0,0 +1,276 @@
+					let record = sonarMeasures[r];
+					let metricName = Object.keys(config.METRIC_MAP).find(key => config.METRIC_MAP[key] === record.metric)[0];
+					output.metrics.push({
+						"name": metricName,
+						"link": config.SERVICE_API_URL + config.HELP_SERVICE + "?metrics=" + metricName,
+						"values": getHistoryValues(record.history)
+					});
+					
+				}
+			} else {
+				for (let n = metrics.length - 1; n >= 0; n--) {
+					let record = metrics[n];
+					output.metrics.push({
+						"name": record,
+						"link": config.SERVICE_API_URL + config.HELP_SERVICE + "?metrics=" + record,
+						"values": []
+					});
+				}
+			}
+
+			resolve(output);
+		}
+	});
+}
+
+function getQuery(serviceContext) {
+	let query = {};
+
+	if (serviceContext.query && serviceContext.query !== {}) {
+		query = Object.keys(svcContext.query).reduce((acc, key) => (acc[key.toLowerCase()] = svcContext.query[key], acc), {});
+	}
+
+	return query;
+}
+
+function getMetrics(query, config, messages) {
+	//if metrics is in query validate the metrics requested against allowed metrics
+	let results = {};
+	let metrics;
+	
+	if (query.metrics) {
+		metrics = query.metrics.split(",");
+		
+		if (metrics.length >= 1) {
+			
+			let allowed_metrics = config.ALLOWED_METRICS.map(v => v.toLowerCase());
+			let invalid_metrics = [];
+
+			for (let j = metrics.length - 1; j >= 0; j--) {
+				if (allowed_metrics.indexOf(metrics[j]) === -1) {
+					invalid_metrics.push(metrics[j]);
+				}
+			}
+
+			if (invalid_metrics.length > 0) {
+				let message = messages.MISSING_METRICS + invalid_metrics.join(", ");
+				result = { "error": message, "metrics": []};
+				return result;
+			}
+		} else {
+			result = { "error": messages.INVALID_METRICS + query.metrics, "metrics": []};
+			return result;
+		}
+	} else {
+		result = { "metrics": config.ALLOWED_METRICS };
+	}
+
+	return result;
+}
+
+function replaceKeys(obj, find, replace) {
+	return Object.keys(obj).reduce(
+		(acc, key) => Object.assign(acc, {
+			[key.replace(find, replace)]: obj[key]
+		}), {}
+	);
+}
+
+function getHistoryValues(valuesArray){
+	return valuesArray.map(obj => replaceKeys(obj, 'date', 'ts'));
+}
+
+// get sonar token
+function getSonarToken(config) {
+	return new Promise((resolve, reject) => {
+		const svcPayload = {
+			uri: config.SERVICE_API_URL + config.TOKEN_URL,
+			method: 'POST',
+			json: {
+				"username": config.SONAR_USER,
+				"password": config.SONAR_PASSWORD
+			},
+			rejectUnauthorized: false
+		};
+		logger.info("Getting Sonar token");	
+		request(svcPayload, (error, response, body) => {
+			if (response && response.statusCode === 200 && body && body.data) {
+				resolve(body.data.token);
+			} else {
+				logger.error(response.body.message);
+				reject({
+					"error": "Could not get authentication token.",
+					"message": response.body.message
+				});
+			}
+		});
+	});
+}
+
+//get auth token
+function getEnvironmentToken(config) {
```

Comment:
Better name for this function is `getAuthToken` or `getJazzAuthToken`

Reply:
you know i thought abt it too and now that you are pointing it out, will update it.